c7n/resources/gamelift.py

```diff
@@ -41,3 +42,5 @@ class resource_type(object):
         dimension = None
         batch_detail_spec = (
             "describe_fleet_attributes", "FleetIds", None, "FleetAttributes")
+        filter_name = 'BuildId'
```

Comment:
that doesn't seem quite right. buildid is a parent for potentially a set of fleets (ie. its a parent).

Reply:
From the boto3 docs the list_fleets does use buildID as a filter:
```
response = client.list_fleets(
    BuildId='string',
    Limit=123,
    NextToken='string'
)
```

and the cloudtrail includes the info:
```
"eventSource": "gamelift.amazonaws.com",
    "eventName": "CreateFleet",
    "awsRegion": "us-east-1",
    "sourceIPAddress": "XXX.XXX.XXX.XXX",
    "userAgent": "signin.amazonaws.com",
    "requestParameters": {
        "buildId": "build-c6e5be7e-c55a-41c1-952b-eba575fab016",
        "eC2InboundPermissions": [
            {
                "ipRange": "0.0.0.0/0",
                "fromPort": 33435,
                "protocol": "UDP",
                "toPort": 33435
            }
        ],
        "eC2InstanceType": "c4.large",
        "serverLaunchPath": "C: \\game\\Bin64.Release.Dedicated\\MultiplayerProjectLauncher_Server.exe",
        "metricGroups": [
            "default"
        ],
        "description": "ThisFleetwascreatedfromtheSamplepage.",
        "serverLaunchParameters": "+sv_port33435+gamelift_start_server",
        "name": "GameLiftSampleFleet: wowcraft"
    },
    "responseElements": {
        "fleetAttributes": {
            "fleetId": "fleet-771b29e6-c085-4c3f-b049-48647931f7d6",
            "serverLaunchPath": "C: \\game\\Bin64.Release.Dedicated\\MultiplayerProjectLauncher_Server.exe",
            "newGameSessionProtectionPolicy": "NoProtection",
            "status": "NEW",
            "creationTime": "Sep22,
20171: 12: 42PM",
            "buildId": "build-c6e5be7e-c55a-41c1-952b-eba575fab016",
            "metricGroups": [
                "default"
            ],
            "operatingSystem": "WINDOWS_2012",
            "description": "ThisFleetwascreatedfromtheSamplepage.",
            "serverLaunchParameters": "+sv_port33435+gamelift_start_server",
            "name": "GameLiftSampleFleet: wowcraft",
            "fleetArn": "arn: aws: gamelift: us-east-1: : fleet/fleet-771b29e6-c085-4c3f-b049-48647931f7d6"
        }
    }
```

Reply:
build id is a build number, if you query on build id, what will you get.. all the fleets that were created with that build id, what was the event for ... one fleet.

another example in the same context, say you wanted to do a policy on create of an ecs service, if you did the same you'd be passing in cluster id, and then evaluating all the services in a cluster including those that already exist, as opposed to the one that was actually evented.